src/common/snippets/src/pass/collapse_subgraph.cpp

```diff
@@ -190,10 +190,13 @@ auto is_supported_op(const std::shared_ptr<const Node> &n) -> bool {
 }
 
 auto has_supported_in_out(const std::shared_ptr<const Node> &n) -> bool {
-    auto supported = [](descriptor::Tensor& t) -> bool {
+    auto supported = [&n](descriptor::Tensor& t) -> bool {
         // TODO [122585] Need to add dynamic rank support
         if (t.get_partial_shape().rank().is_dynamic())
             return false;
+        // If Convert node has unsupported data type, it should not be tokenized.
+        if (ov::is_type<const opset1::Convert>(n))
+               return TokenizeSnippets::get_supported_element_types().count(t.get_element_type()) != 0;
```

Comment:
It should be handled by plugin callback with supported precisions. Could you tell me please which test failed without this check?

Reply:
Appied. Removed Convert check, and removed setting IGNORE_CALLBACK for Select and BroadcastSelect Setup() in Snippets tests, as we've discussed offline. Beside some additional revisions are made to solve related test failure on x64.
1. Expected number of subgraphs is set to be 0 for smoke_Snippets_Select_Dynamic and smoke_Snippets_BroadcastSelect_Dynamic. Because  set by plugin callback (as callback is not agnored now), Dynamic layers are not tokenized https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp#L820.
2. As Broadcast  is also not tokenized https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp#L834-L835, the expected number of nodes for smoke_Snippets_BroadcastSelect tests should be 4 (2 Convert + 1 Broadcast + 1 Select).
3. When inShape0 and inShape3 both equal to {1, 8, 2, 1} https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/tests/functional/shared_tests_instances/snippets/select.cpp#L49-L52, there is actually no Broadcast layer in the test model. It seems no need to retain such cases. Besides, as Broadcast layer is not there, the expect nodes should 2(1 Convert + 1 Select) for such smoke_Snippets_BroadcastSelect test cases. To avoid such diverged test behavior, I simply changed inShape0 to be {1, 8, 1, 1}.
Please feel free to have further discussions, if you have concerns. Thanks Alexandra!

Reply:
Yeah, I see. Thank you for exploration!

As I said offline, since ARM doesn't support broadcast and select yet, you can just disable the corresponding tests in `skip_test_config`. So you don't need to update select related tests.

Thus, I suggest to disable all tests that contains unsupported ops by ARM (Select, Broadcast and other) and return ` IGNORE_CALLBACK` to Select tests

Reply:
Oh, in fact these are x64 issues, that emerge because we move conditions for tokenization from snippets lib to plugin. The arm test cases have already been skipped by https://github.com/xuchen-intel/openvino/blob/feature/arm_snippets/src/plugins/intel_cpu/tests/functional/shared_tests_instances/skip_tests_config.cpp#L342. Thanks Alexandra!

Reply:
Yeah, so we should revert Select related changes (return IGNORE_CALLBACK, shapes)
Because as I said before, we use `IGNORE_CALLBACK` to test some use-case that can be disabled in common pipeline on models.
For example, it's dynamism.  At the moment CPU Plugin disabled snippets tokenization of dynamic nodes to avoid perf degradations due to kernel recompilation.
Or we support Broadcast only with Select in MHA pattern. However, we test this use-case as mini subgraph `Broadcast+Select` to have better test coverage.
To validate these cases, we should have `IGNORE_CALLBACK` in test body

Reply:
Thanks Alexandra for the offline dicussion and your great approach! :fire:  The IGNORE_CALLBACK is applied in specific callback checks.